exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/ConvertCountToDirectScan.java

```diff
@@ -154,7 +154,7 @@ public void onMatch(RelOptRuleCall call) {
   private Map<String, Long> collectCounts(PlannerSettings settings, DrillAggregateRel agg, DrillScanRel scan, DrillProjectRel project) {
     final Set<String> implicitColumnsNames = ColumnExplorer.initImplicitFileColumns(settings.getOptions()).keySet();
     final GroupScan oldGrpScan = scan.getGroupScan();
-    final long totalRecordCount = oldGrpScan.getScanStats(settings).getRecordCount();
+    final long totalRecordCount = (long)oldGrpScan.getScanStats(settings).getRecordCount();
```

Comment:
getRecordCount() (in ScanStats.java) already returns a long, so a cast here is not needed ....